hg-fast-export.py

```diff
@@ -196,8 +196,9 @@ def refresh_gitmodules(ctx):
       gitmodules+=refresh_hg_submodule(name,subrepo_info)
 
   if len(gitmodules):
+    gitmodules += b'\n'
```

Comment:
This is redundant (see the review comment)

Reply:
It's not redundant because it allows the removal of the `+1` in `len(gitmodules)+1`, but then `wr()` adds an extra newline, when it should be `wr_no_nl()`.

I've fixed this and pushed a new pull request.

Reply:
The extra `\n` is not bad, it increases the readability of the dump. And in this case it saved us when when we told git that the data was one byte larger than it was.

Reply:
But it's not just adding a newline on the stream, it's adding a newline on the file. If you want an extra newline, then probably `wr()` should be used instead of `wr_no_nl()`, but you can't add arbitrary newlines, care must be taken to only add one. Which is why `wr_data()` makes sense.

Reply:
> it's adding a newline on the file

Yes, that's why I said it was redundant as the file already ends with a `\n`. We drop the `+1` here, and the new `wr_data()` correctly writes the data and preserves the readability. The extra `\n` added by `wr()` was what saved us from the effects of the erroneous `+1` as the inadvertently  added `\n` to `.gitmodules` is harmless and fast-import accepted our output despite it being conceptually wrong.